### ISSUE TITLE
Learning examples (test/train): can training examples and test examples come from different nodes? or even different base types?  

### ISSUE THREAD

**Issue by kordjamshidi** [IllinoisCogComp/saul#109]
In the current Saul, we assume the learners get test and train examples from same specific node  of a same base type T, the training and test instances are given to the learner via the related node, but the learner only cares about T to be able to get the features ( using property queries) and to get the label for each test/train example. But in practice there are cases that the reader does not give us all examples we need, this is particularly the case with the negative examples in case of structured output prediction models.  We might use a different type of node to generate negative examples (or in general candidate) for another type of node while the positive examples might come directly form the reader.  Can we provide a principled way for dealing with such a situation to the application programmer?

**Comment by danyaljj** [comment]
@kordjamshidi did you address this while writing SRL?

**Comment by kordjamshidi** [comment]
no, we used TextAnnotation and Xupalmer relations so , we did not need to address this after all but I think important to keep it.
